src/main/groovy/io/seqera/wave/service/blob/impl/BlobCacheServiceImpl.groovy

```diff
@@ -194,6 +202,12 @@ class BlobCacheServiceImpl implements BlobCacheService {
                     ? blobConfig.statusDuration
                     : blobConfig.statusDelay.multipliedBy(10)
 
+            //check if the cached blob size is correct
+            if(info.contentLength != getBlobSize(route.targetPath)){
+                log.warn("Blob size mismatch for object '${info.locationUri}'")
+                CompletableFuture.supplyAsync(() -> deleteBlob(route.targetPath), executor)
+            }
```

Comment:
This should be done only `if( result.succeeded() )` \*and* the result should be modified to report a failure e.g. `result.failed("Blob uploaded does not match the expected size")`

Reply:
done

Reply:
there should be a test that when the uploaded size does not match the result status is failed.

Reply:
I mean, corresponding this change https://github.com/seqeralabs/wave/pull/533/commits/d62b2ceb1285bac4da40a4a4cc677c4fc955c489

Reply:
ok understood

Reply:
done